### PR TITLE
pkg/trace/api: Add support for 128-Bit TraceIds in OpenTelemetry integration.

### PR DESCRIPTION
**Pull request by dinooliva**
### What does this PR do?

Updates the translation from OTLP to Datadog trace data in the trace agent to support the full 128-bits of the TraceId by hex-encoding the higher-order 64 bits and adding it as the `_dd.p.tid` tag to the first span of the trace chunk, as defined in the "Supporting 128-Bit TraceIds - Integrations" RFC.

### Motivation

This change is being made as part of the ongoing effort to support 128-bit TraceIds throughout the organization.

### Possible Drawbacks / Trade-offs

This change will add 1 tag with a 16 character value to each span chunk coming from OpenTelemetry services, the overall impact should be minor (note that the Datadog tracer libraries are adding the same tag).

### Describe how to test/QA your changes

Send a trace from an OpenTelemetry service to the trace agent's OTLP endpoint and verify that the higher-order 64 bits of the TraceId is encoded in the `_dd.p.tid` tag of the first span of the trace chunk for that trace.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by dinooliva** [comment]
My apologies, I just realized I didn't prepend 'pkg/trace/api: ' to my commit comments, I'll take care to do that moving forward.

File: pkg/trace/api/otlp.go [hunk 1]
```diff
@@ -325,6 +330,9 @@ func (o *OTLPReceiver) createChunks(tracesByID map[uint64]pb.Trace, prioritiesBy
 			Tags:  map[string]string{"_dd.otlp_sr": rate},
 			Spans: spans,
 		}
+		if tid, ok := tidByID[k]; ok {
+			traceutil.SetMeta(spans[0], "_dd.p.tid", tid)
```
Comment (gbbr) [inline pkg/trace/api/otlp.go:334]: We already have `otel.trace_id` on [each and every span](https://github.com/DataDog/datadog-agent/blob/055d626c2415fa37ffb9968b6208eef52a3a8338/pkg/trace/api/otlp.go#L451). Why do we need to duplicate it in another tag (despite the fact that we cut off the second half)? Can't the backend just pick it up? This seems like unnecessary complexity...

**Review by maycmlee — APPROVED** [review]
👍 for docs

**Comment by dinooliva** [comment]
After further discussions, we will add support for the `otel.trace_id` tag in the backend to provides 128-bit TraceIds via this path.

**Pull request closed without merging**